src/fileformat/file_format/pe/pe_format.cpp

```diff
@@ -3350,5 +3409,320 @@ const VisualBasicInfo* PeFormat::getVisualBasicInfo() const
 	return &visualBasicInfo;
 }
 
+/**
+ * Scan for file format anomalies
+ */
+void PeFormat::scanForAnomalies()
+{
+	scanForSectionAnomalies();
+	scanForResourceAnomalies();
+	scanForImportAnomalies();
+	scanForExportAnomalies();
+	scanForOptHeaderAnomalies();
+	scanForObsoleteCharacteristics();
+}
+
+/**
+ * Scan for section anomalies
+ */
+void PeFormat::scanForSectionAnomalies()
+{
+	std::size_t nSecs = getDeclaredNumberOfSections();
+	
+	const PeCoffSection *epSec = dynamic_cast<const PeCoffSection*>(getEpSection());
+	if (epSec)
+	{
+		// scan EP in last section
+		const PeCoffSection *lastSec = (nSecs) ? getPeSection(nSecs - 1) : nullptr;
+		if (epSec == lastSec)
+		{
+			anomalies.emplace_back(std::make_pair<std::string, std::string>("epInLastSec",
```

Comment:
Two remarks concerning the use of `std::make_pair()` and `emplace_back()`:
1. When using `std::make_pair`, you are not supposed to specify the template parameters (see e.g. [this SO answer](https://stackoverflow.com/a/9642232/2580955)). The whole point of `std::make_pair()` is that you don't have to specify the template parameters. Otherwise, you could just as well create `std::pair` directly.
2. There is no benefit of using `emplace_back()` with r-values (you could call just `push_back()`). The benefit of `emplace_back()` comes from the fact that you can pass just the arguments to the constructor of the objects that a vector holds, which both simplifies the code and possibly makes it faster, as the object is created in-place.

Thus, instead of
```cpp
anomalies.emplace_back(std::make_pair<std::string, std::string>("x", "y"));
```
write just
```cpp
anomalies.emplace_back("x", "y");
```